common/src/binary/BinaryBuffer.cpp

```diff
@@ -0,0 +1,215 @@
+#include <rcbe-engine/binary/BinaryBuffer.hpp>
+
+namespace {
+template <typename ArrayBasedContainer>
+size_t compute_overall_size(const ArrayBasedContainer &abc) {
+    return std::accumulate(abc.begin(), abc.end(), 0, [](auto sum, const auto& entry) {
+        return sum + entry.size();
+    });
+}
+}
+
+namespace rcbe::binary {
+BinaryBuffer::BinaryBuffer(std::istream& is) {
+    read(is, buffer_);
+}
+
+BinaryBuffer::BinaryBuffer(std::vector<ByteType>&& buf)
+:
+buffer_(std::move(buf))
+{}
+
+BinaryBuffer::BinaryBuffer(std::initializer_list<BinaryBuffer> &&bb) {
+    buffer_.reserve(buffer_.size() + compute_overall_size(bb));
+    for (auto &b : bb) {
+        append(std::move(b));
+    }
+}
+
+BinaryBuffer::BinaryBuffer(std::vector<BinaryBuffer> &&bb) {
+    buffer_.reserve(buffer_.size() + compute_overall_size(bb));
+    for (const auto &b : bb) {
+        append(bb);
+    }
+}
+
+BinaryBuffer &BinaryBuffer::operator=(const BinaryBuffer &bb) {
+    (void)append(bb);
+    return *this;
+}
+
+BinaryBuffer &BinaryBuffer::operator=(BinaryBuffer &&bb) {
+    (void)append(std::move(bb));
+    return *this;
+}
+
+BinaryBuffer &BinaryBuffer::append(const BinaryBuffer &bb) {
+    buffer_.reserve(buffer_.size() + bb.buffer_.size());
+    buffer_.insert(buffer_.end(), bb.buffer_.begin(), bb.buffer_.end());
+    return *this;
+}
+
+BinaryBuffer &BinaryBuffer::append(BinaryBuffer &&bb) {
+    buffer_.reserve(buffer_.size() + bb.buffer_.size());
+    buffer_.insert(buffer_.end(), bb.buffer_.begin(), bb.buffer_.end());
```

Comment:
Both implementation have the same code. Also these two don't take into account a view case.